data_upload/extract.py

```diff
@@ -0,0 +1,182 @@
+
+COLUMNS = ['place', 'time', 'magnitude', 'alert_type', 'felt_report_count',
+           'cdi', 'latitude', 'longitude', 'depth', 'magnitude_type',
+           'network_name']
+
+PDF_FILE = f"""/tmp/{datetime.today().strftime('%Y-%m-%d')}-data.pdf"""
+
+
+def get_connection() -> connection:
+    """Gets connection to database"""
+    return psycopg2.connect(
+        user=os.getenv("DB_USER"),
+        password=os.getenv("DB_PASSWORD"),
+        host=os.getenv("DB_HOST"),
+        port=os.getenv("DB_PORT"),
+        database=os.getenv("DB_NAME")
+    )
+
+
+COLUMN_NAME_MAP = {
+    'time': 'Time',
+    'place': 'Place',
+    'magnitude': 'Magnitude',
+    'felt_report_count': 'Felt Report Count',
+    'cdi': 'CDI',
+    'latitude': 'Latitude',
+    'longitude': 'Longitude',
+    'depth': 'Depth',
+    'alert_type': 'Alert Type',
+    'magnitude_type': 'Magnitude Type',
+    'network_name': 'Network Name',
+}
+
+
+def extract_data() -> pd.DataFrame:
+    """Extracts data from the database and prepares it for PDF generation"""
+    conn = None
+    try:
+        conn = get_connection()
+        logging.info("Executing query...")
+        earthquakes = pd.read_sql(QUERY, conn)
+        earthquakes = earthquakes[COLUMNS]
+        earthquakes = earthquakes.rename(columns=COLUMN_NAME_MAP)
+        earthquakes['Depth'] = earthquakes['Depth'].apply(
+            lambda x: f"{x:.2f}" if pd.notnull(x) else x)
+        earthquakes['Latitude'] = earthquakes['Latitude'].apply(
+            lambda x: f"{x:.6f}" if pd.notnull(x) else x)
+        earthquakes['Longitude'] = earthquakes['Longitude'].apply(
+            lambda x: f"{x:65f}" if pd.notnull(x) else x)
+        earthquakes["Time"] = earthquakes["Time"].apply(
+            lambda x: pd.to_datetime(x).strftime(
+                "%Y-%m-%d %H:%M") if pd.notnull(x) else x
+        )
+
+        return earthquakes
+
+    except Exception as e:
+        logging.error("Error during data extraction: %s", e)
+        raise
+    finally:
+        if conn:
+            conn.close()
+
+
+def make_pdf(data: pd.DataFrame) -> None:
+    """Generates a PDF from the provided dataframe"""
+    logging.info("Writing data to PDF: %s", PDF_FILE)
+    styles = getSampleStyleSheet()
+    header_styles = styles['BodyText']
+    header_styles.fontName = "Helvetica-Bold"
+
+    table_data = [
+        [Paragraph(str(col), header_styles) for col in data.columns]
+    ]
+    for row in data.values.tolist():
+        wrapped_row = [Paragraph(str(cell), styles["BodyText"])
+                       for cell in row]
+        table_data.append(wrapped_row)
+
+    pdf = SimpleDocTemplate(PDF_FILE, pagesize=landscape(letter))
+
+    col_widths = [
+        170,  # place
+        80,   # time
+        65,   # magnitude
+        40,   # alert type
+        50,   # felt report count
+        40,   # cdi
+        70,   # latitude
+        70,   # longitude
+        42,   # depth
+        64,   # magtype
+        52,   # network name
+    ]
```

Comment:
maybe make this a global variable aswell?

Reply:
good idea! didn't think of that